src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -280,89 +251,40 @@ export function getDiscoverStateContainer({
     internalStateContainer.transitions.setSavedDataViews(dataViewList);
   };
 
+  const loadAndResolveDataView = async (id: string, actualSavedSearch: SavedSearch) => {
+    const nextDataViewData = await loadDataView(services.dataViews, services.uiSettings, id);
+    const nextDataView = resolveDataView(
+      nextDataViewData,
+      actualSavedSearch.searchSource,
+      services.toastNotifications
+    );
+    return { fallback: !nextDataViewData.stateValFound, dataView: nextDataView };
+  };
+
   return {
     kbnUrlStateStorage: stateStorage,
-    appState: appStateContainerModified,
+    appState: appStateContainer,
     internalState: internalStateContainer,
     dataState: dataStateContainer,
     searchSessionManager,
     startSync: () => {
-      const { start, stop } = syncAppState();
+      const { start, stop } = appStateContainer.syncState();
       start();
       return stop;
     },
-    setAppState: (newPartial: AppState) => setState(appStateContainerModified, newPartial),
+    setAppState: (newPartial: AppState) => setState(appStateContainer, newPartial),
```

Comment:
In what situations would we want to call `setAppState` instead of `appStateContainer.update`? They seem to do the same thing except `appStateContainer.update` also sets the previous state. Related to the comment above, but I feel like this could get confusing, and just calling `appStateContainer.update` would probably be preferred.

Reply:
yes, another potential cleanup, I would target this when moving more functions away from the main state container